### ISSUE TITLE
Change CLI and Node.js to check `.css` files by default

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#7409]
### What is the problem you're trying to solve?

Check `.css` files by default for consistent behaviour with ESLint's [logical defaults](https://eslint.org/blog/2022/08/new-config-system-part-2/#the-goals-of-flat-config).

It would be a breaking change and could happen in `17.0.0`.

Blocked by:

- https://github.com/stylelint/stylelint/issues/3860

Refs: 

- https://github.com/stylelint/stylelint/issues/7408
- https://github.com/stylelint/stylelint/issues/6689#issuecomment-1465130308
- https://github.com/stylelint/stylelint/issues/3411

### What solution would you like to see?

To check `**/*.css` files by default for the:

- CLI when `npx stylelint` is used, rather than show help
- Node.js API when neither `files` nor `code` are set 

For the latter, it'd clean up [the "optional" behaviour](https://stylelint.io/user-guide/node-api#files):

> Though both `files` and `code` are "optional", you must have one and cannot have both.

**Comment by ybiquitous** [comment]
@jeddy3 Thanks for opening this proposal. I agree with this idea, which should improve user experience. 👍🏼 

I have one question: If `files[]` in a config (#3860) were present, the default `**/*.css` would change, right? E.g.

```js
// Config
export default {
  files: ["**/*.scss"],
  customSyntax: "postcss-scss",
  rules: { /*...*/ }
};
```

```shell
# Will scan **/*.scss files, instead of the default **/*.css files
stylelint
```

**Comment by jeddy3** [comment]
That's correct.

**Comment by jeddy3** [comment]
(@Mouvedia [label conventions](https://stylelint.io/maintainer-guide/issues) - we typically remove the `status: need *` label when we've agreed on a cause of action.)

**Comment by Mouvedia** [comment]
I was planning to assign myself the blocker after I am done with a POC. I was too eager; I will wait for the agreement.

**Comment by ybiquitous** [comment]
If we implement this issue, it will be included in the next major version, right?

**Comment by jeddy3** [comment]
That's correct. It'd be breaking change. We should only start to work on it near the time we're planning to release `17.0.0` so that we don't have a long-lived branch.
